source/adapters/native_cpu/enqueue.cpp

```diff
@@ -247,11 +261,7 @@ UR_APIEXPORT ur_result_t UR_APICALL urEnqueueKernelLaunch(
   if (phEvent) {
     *phEvent = event;
   }
-  event->set_callback([hKernel, event]() {
-    event->tick_end();
-    // TODO: avoid calling clear() here.
-    hKernel->_localArgInfo.clear();
-  });
+  event->set_callback([event]() { event->tick_end(); });
```

Comment:
Why was the `_localArgInfo.clear();` here? Was it a workaround for another issue? Is whatever reason it had for being there being addressed in a different way after your PR?